TeachingRecordSystem/tests/TeachingRecordSystem.SupportUi.Tests/PageTests/Persons/PersonDetail/InductionTests.cs

```diff
@@ -0,0 +1,232 @@
+        var setCompletionDate = DateOnly.FromDateTime(DateTime.Now);
+        var person = await TestData.CreatePersonAsync(
+                x => x
+                .WithQts()
+                .WithInductionStatus(builder => builder
+                    .WithStatus(setInductionStatus)
+                    .WithStartDate(setStartDate)
+                    .WithCompletedDate(setCompletionDate)
+                ));
+
+        var request = new HttpRequestMessage(HttpMethod.Get, $"/persons/{person.ContactId}/induction");
+
+        // Act
+        var response = await HttpClient.SendAsync(request);
+
+        // Assert
+        var doc = await AssertEx.HtmlResponseAsync(response);
+        var inductionStatus = doc.GetElementByTestId("induction-status");
+        Assert.Contains(StatusStrings[setInductionStatus], inductionStatus!.TextContent);
+        var completionDate = doc.GetElementByTestId("induction-completion-date")!.Children[1].TextContent;
+        Assert.Contains(setCompletionDate.ToString("d MMMM yyyy"), completionDate);
+    }
+
+    //[Theory]
+    //[InlineData(InductionStatus.InProgress)]
+    //[InlineData(InductionStatus.Passed)]
+    //[InlineData(InductionStatus.Failed)]
+    // CL - what I had inferred from page designs
+    //public async Task Get_WithPersonIdForPersonWithInductionStatusMangedByCPD_DisplaysWarning(InductionStatus setInductionStatus)
+    //{
+    //    // Arrange
+    //    var expectedWarning = "To change a teacher's induction status ";
+    //    var setStartDate = DateOnly.FromDateTime(DateTime.Now);
+    //    var person = await TestData.CreatePersonAsync(
+    //            x => x
+    //            .WithQts()
+    //            .WithInductionStatus(builder => builder
+    //                .WithStatus(setInductionStatus)
+    //                .WithStartDate(setStartDate)
+    //            ));
+
+    //    var request = new HttpRequestMessage(HttpMethod.Get, $"/persons/{person.ContactId}/induction");
+
+    //    // Act
+    //    var response = await HttpClient.SendAsync(request);
+
+    //    // Assert
+    //    var doc = await AssertEx.HtmlResponseAsync(response);
+    //    Assert.Contains(expectedWarning, doc.GetElementByTestId("induction-status-warning")!.TextContent);
+    //}
+
+    //[Fact]
+    //// replacement for above, when TestData builder methods are there - plus add tests for these conditions not met
+    //public async Task Get_WithPersonIdForPersonWithInductionStatusMangedByCPD_DisplaysWarning()
+    //{
+    //    //Arrange
+    //    var expectedWarning = "To change a teacher's induction status ";
+    //    var sevenYearsAgo = DateTime.Now.AddDays(-1).AddYears(-7); // more than 7 years ago
+    //    var setCompletionDate = DateOnly.FromDateTime(sevenYearsAgo);
+    //    var person = await TestData.CreatePersonAsync(
+    //            x => x
+    //        .WithQts()
+    //        .WithCpdInductionStatus(builder => builder
```

Comment:
This needs adding to the TestData.CreatePerson builder?

Reply:
> This needs adding to the TestData.CreatePerson builder?

It does, for now something like this will work in its place:

```cs
        var person = await TestData.CreatePersonAsync(...);

        await WithDbContextAsync(async dbContext =>
        {
            dbContext.Attach(person.Person);

            person.Person.SetCpdInductionStatus(
                InductionStatus.InProgress,
                startDate,
                completedDate: null,
                cpdModifiedOn: Clock.UtcNow,
                updatedBy: SystemUser.SystemUserId,
                now: Clock.UtcNow,
                out _);

            await dbContext.SaveChangesAsync();
        });
```

Reply:
ta - done